### PR TITLE
feat: autocontrolled `useTable` hook

### PR DESCRIPTION
**Pull request by ling1726**
`useTable` hook now supports autoncontrolling sort and selection states

## Current Behavior

`useTable` is completely uncontrolled

## New Behavior

`useTable` can be controlled and uncontrolled, added extra stories to that effect

## Related Issue(s)

Addresses #24226

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/ff6eaee1c2f0702585ec8b5d6545e564a0ef5a86'>ff6eaee1c2f0702585ec8b5d6545e564a0ef5a86</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/Avatar-mount_base.html">1324</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/Avatar-mount.html">1340</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/Button-mount_base.html">945</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/Button-mount.html">954</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/FluentProvider-mount_base.html">1570</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/FluentProvider-mount.html">1582</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/FluentProviderWithTheme-mount_base.html">622</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/FluentProviderWithTheme-mount.html">622</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">593</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">589</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">636</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">632</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/MakeStyles-mount_base.html">1901</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/MakeStyles-mount.html">1927</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/SpinButton-mount_base.html">2530</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24688/perf-test-react-components/SpinButton-mount.html">2540</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-table/src/components/Table/Table.types.ts [hunk 1]
```diff
@@ -26,19 +21,7 @@ export type TableContextValues = {
...
-    onSortColumnChange?: (
-      e: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>,
-      data: { sortState: SortState },
-    ) => void;
-
-    sortState?: SortState;
-
-    defaultSortState?: SortState;
-  };
+export type TableProps = ComponentProps<TableSlots> & {} & Partial<TableContextValue>;
```
Comment (bsunderhus) [inline packages/react-components/react-table/src/components/Table/Table.types.ts:24]:
```suggestion
export type TableProps = ComponentProps<TableSlots> & Partial<TableContextValue>;
```

Do we need this `{}`? 😅

**Review by bsunderhus — APPROVED** [review]

File: packages/react-components/react-table/src/components/Table/Table.types.ts (hunk 1 above)
Comment (ling1726) [reply packages/react-components/react-table/src/components/Table/Table.types.ts:24]: nope

**Pull request merged**